Description: MinGW: Cannot export (anonymous namespace)::thread_binder
 Upstream changeset [76598]. 
Author: viboes
Bug: 671409
Forwarded: https://svn.boost.org/trac/boost/ticket/6431


trunk/libs/thread/test/util.inl

Index: trunk/libs/thread/test/util.inl
===================================================================
--- a/libs/thread/test/util.inl
+++ b/libs/thread/test/util.inl
@@ -99,5 +99,7 @@
     int secs;
 };
-
+}
+namespace thread_detail_anon
+{
 template <typename F>
 class indirect_adapter
@@ -127,4 +129,9 @@
 };
 
+}
+// boostinspect:nounnamed
+namespace 
+{
+
 template <typename F>
 void timed_test(F func, int secs,
@@ -132,5 +139,5 @@
 {
     execution_monitor monitor(type, secs);
-    indirect_adapter<F> ifunc(func, monitor);
+    thread_detail_anon::indirect_adapter<F> ifunc(func, monitor);
     monitor.start();
     boost::thread thrd(ifunc);
@@ -139,4 +146,9 @@
 }
 
+}
+
+namespace thread_detail_anon
+{
+
 template <typename F, typename T>
 class thread_binder
@@ -152,9 +164,18 @@
 };
 
+}
+
+// boostinspect:nounnamed
+namespace 
+{
 template <typename F, typename T>
-thread_binder<F, T> bind(const F& func, const T& param)
-{
-    return thread_binder<F, T>(func, param);
-}
+thread_detail_anon::thread_binder<F, T> bind(const F& func, const T& param)
+{
+    return thread_detail_anon::thread_binder<F, T>(func, param);
+}
+}
+
+namespace thread_detail_anon
+{
 
 template <typename R, typename T>
@@ -173,9 +194,13 @@
 };
 
-
+}
+
+// boostinspect:nounnamed
+namespace 
+{
 template <typename R, typename T>
-thread_member_binder<R, T> bind(R (T::*func)(), T& param)
-{
-    return thread_member_binder<R, T>(func, param);
+thread_detail_anon::thread_member_binder<R, T> bind(R (T::*func)(), T& param)
+{
+    return thread_detail_anon::thread_member_binder<R, T>(func, param);
 }
 } // namespace
